cms/core/migrations/0006_update_user_groups.py

```diff
@@ -0,0 +1,182 @@
+    Collection = apps.get_model("wagtailcore.Collection")
+    GroupCollectionPermission = apps.get_model("wagtailcore.GroupCollectionPermission")
+
+    root_collection = Collection.objects.get(depth=1)
+
+    # Grant add, change and delete permissions to Publishing Admins
+    for permission in WAGTAIL_PERMISSION_TYPES:
+        GroupCollectionPermission.objects.create(
+            group=Group.objects.get(name=settings.PUBLISHING_ADMINS_GROUP_NAME),
+            collection=root_collection,
+            permission=Permission.objects.get(content_type__app_label="wagtailimages", codename=f"{permission}_image"),
+        )
+
+    # Grant "choose" permission to Publishing Admins and Publishing Officers
+    for group_name in [settings.PUBLISHING_ADMINS_GROUP_NAME, settings.PUBLISHING_OFFICERS_GROUP_NAME]:
+        group = Group.objects.get(name=group_name)
+        GroupCollectionPermission.objects.create(
+            group=group,
+            collection=root_collection,
+            permission=Permission.objects.get(content_type__app_label="wagtailimages", codename="choose_image"),
+        )
+
+
+def create_snippet_permissions(apps):
+    """Allow Publishing Admins to create, edit and delete snippets."""
+    snippet_classes = [
+        # app, model, add publish permission
+        ("core", "GlossaryTerm", False),
+        ("core", "ContactDetails", False),
+        ("navigation", "MainMenu", True),
+        ("navigation", "FooterMenu", True),
+    ]
+
+    for app, model, add_publish_permission in snippet_classes:
+        for permission in WAGTAIL_PERMISSION_TYPES:
+            assign_permission_to_group(
+                apps,
+                settings.PUBLISHING_ADMINS_GROUP_NAME,
+                permission_codename=f"{permission}_{model.lower()}",
+                app=app,
+                model=model,
+            )
+        if add_publish_permission:
+            assign_permission_to_group(
+                apps,
+                settings.PUBLISHING_ADMINS_GROUP_NAME,
+                permission_codename=f"publish_{model.lower()}",
+                app=app,
+                model=model,
+            )
+
+
+def create_bundle_permissions(apps):
+    """Grant all permissions on Bundles to Publishing Admins and Publishing Officers,
+    and view-only access to Viewers.
+    """
+    bundles_app = "bundles"
+    bundle_class = "Bundle"
+
+    for permission in WAGTAIL_PERMISSION_TYPES:
+        for group_name in [settings.PUBLISHING_ADMINS_GROUP_NAME, settings.PUBLISHING_OFFICERS_GROUP_NAME]:
+            assign_permission_to_group(
+                apps,
+                group_name=group_name,
+                permission_codename=f"{permission}_{bundle_class.lower()}",
+                app=bundles_app,
+                model=bundle_class,
+            )
+
+    # View-only access for the Viewers group
+    assign_permission_to_group(apps, settings.VIEWERS_GROUP_NAME, "view_bundle", bundles_app, bundle_class)
```

Comment:
can you add it to the other two roles please?

Reply:
Sure, but I'm not entirely clear why we'd need that as the `add` and `change` permissions imply that they can view it (and looking at the `Bundles` page in the UI the PA and PO still have the `Inspect` option for each Bundle).
Is this just for the sake of being explicit?

Reply:
>  Is this just for the sake of being explicit?

Yes